docker/tarfile/src.go

```diff
@@ -373,6 +373,12 @@ func (s *Source) GetBlob(ctx context.Context, info types.BlobInfo) (io.ReadClose
 		if err != nil {
 			return nil, 0, err
 		}
+		succeeded := false
+		defer func() {
+			if !succeeded {
```

Comment:
Why not define return err error on the call and then just check if err != nil?

Reply:
That’s elegant :) This particular method has another successful exit path (`info.Digest == s.configDigest`), so a bit risky.

For the same reason, the `succeeded` variable name is inaccurate, so I’ve at least renamed it to be right.